partiql-spi/src/main/kotlin/org/partiql/spi/connector/ConnectorMetadata.kt

```diff
@@ -56,10 +37,55 @@ public interface ConnectorMetadata {
      * As another example, consider an object within a Namespace that may be a Struct with nested attributes. A user could
      * call [getObjectHandle] with the [path] of "a"."b"."c"."Object"."x". In the Namespace, only object "Object" exists.
      * Therefore, this method will return a [ConnectorObjectHandle] with the "Object" representation and the matching
-     * path: "a"."b"."c"."Object". The returned [ConnectorObjectHandle.absolutePath] must be correct for correct
+     * path: "a"."b"."c"."Object". The returned [ConnectorObjectHandle.path] must be correct for correct
      * evaluation.
      *
      * If the [path] does not correspond to an existing [ConnectorObject], implementers should return null.
      */
-    public fun getObjectHandle(session: ConnectorSession, path: BindingPath): ConnectorObjectHandle?
+    public fun getObjectHandle(path: BindingPath): ConnectorObjectHandle?
```

Comment:
While we currently don't allow for users to change their authorization, almost all of the privileges in SQL1999 are described by the CURRENT_USER (distinct from the SESSION_USER). In the future if we need to support changing the current user, we might need to re-add the ConnectorSession as a parameter.

Reply:
The session as a parameter is redundant. The factory method for ConnectorMetadata has a signature. The ConnectorMetadata would close over this.

```kotlin

// interface method in connector
fun getConnectorMetadata(session: ConnectorSession) : ConnectorMetadata

// session is associated to the metadata
class MyConnectorMetadata(private val session: ConnectorSession) : ConnectorMetadata {

     fun getObjectHandle(path: BindingPath) { ... do somethign with session }

}

```

How it's factored now is redundant, and it allows passing a different session that the one the metadata was created for which is potentially problematic.

```kotlin
val md = getConnectorMetadata(sessionFoo)

md.getObjectHandle(sessionBar, myPath) // !!! 
```